### PR TITLE
Precompile action_cable.js

### PR DESCRIPTION
**Pull request by javan**
Alternative to #23324 
- Adds `assets:compile` rake task to create `dist/action_cable.js`. Apps that (continue to) use `//= require action_cable` will load this precompiled version.
- Allows loading the uncompiled bundle with `//= require action_cable/source` making it possible use unreleased changes in master that haven't yet been released in precompiled form.

### CODE REVIEW

File: actioncable/actioncable.gemspec [hunk 1]
```diff
@@ -24,9 +24,12 @@ Gem::Specification.new do |s|
   s.add_dependency 'nio4r',            '~> 1.2'
   s.add_dependency 'websocket-driver', '~> 0.6.1'
 
-  s.add_development_dependency 'em-hiredis', '~> 0.3.0'
+  s.add_development_dependency 'coffee-script',        '~> 2.4.1'
+  s.add_development_dependency 'coffee-script-source', '~> 1.10.0'
```
Comment (maclover7) [inline actioncable/actioncable.gemspec]: Why do we need this gem? Looks like it's not required anywhere.

Comment (javan) [inline actioncable/actioncable.gemspec]: Sprockets [autoloads them](https://github.com/rails/sprockets/blob/2d30e546872f6f4c5361fe19c2d89b15c5db2dd4/lib/sprockets/autoload.rb#L5)

Comment (maclover7) [inline actioncable/actioncable.gemspec]: ok

Comment (javan) [inline actioncable/actioncable.gemspec]: Sorry, I didn't realize which gem you were asking about. The `coffee-script` gem [depends on `coffee-script-source`](https://github.com/josh/ruby-coffee-script/blob/master/coffee-script.gemspec#L19), but doesn't declare a version. Specifying one here ensures the `.coffee` files are compiled consistently.

**Comment by matthewd** [comment]
This was pulled into #23369, then further adjusted in d6f2000a67cc63aa67414c75ce77de671824ec52

**Pull request closed without merging**